app/code/Magento/Sales/Model/Order/ShipmentDocumentFactory/ExtensionAttributesProcessor.php

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Sales\Model\Order\ShipmentDocumentFactory;
+
+use Magento\Framework\Reflection\ExtensionAttributesProcessor as AttributesProcessor;
+use Magento\Sales\Api\Data\ShipmentCreationArgumentsExtensionInterface;
+use Magento\Sales\Api\Data\ShipmentCreationArgumentsInterface;
+use Magento\Sales\Api\Data\ShipmentExtensionFactory;
+use Magento\Sales\Api\Data\ShipmentExtensionInterface;
+use Magento\Sales\Api\Data\ShipmentInterface;
+
+/**
+ * Build and set extension attributes for shipment.
+ */
+class ExtensionAttributesProcessor
+{
+    /**
+     * @var ShipmentExtensionFactory
+     */
+    private $shipmentExtensionFactory;
+
+    /**
+     * @var ExtensionAttributesProcessor
+     */
+    private $extensionAttributesProcessor;
+
+    /**
+     * @param ShipmentExtensionFactory $shipmentExtensionFactory
+     * @param AttributesProcessor $extensionAttributesProcessor
+     */
+    public function __construct(
+        ShipmentExtensionFactory $shipmentExtensionFactory,
+        AttributesProcessor $extensionAttributesProcessor
+    ) {
+        $this->shipmentExtensionFactory = $shipmentExtensionFactory;
+        $this->extensionAttributesProcessor = $extensionAttributesProcessor;
+    }
+
+    /**
+     * @param ShipmentInterface $shipment
+     * @param ShipmentCreationArgumentsInterface $arguments
+     * @return void
+     */
+    public function execute(
+        ShipmentInterface $shipment,
+        ShipmentCreationArgumentsInterface $arguments = null
+    ): void {
+        if (null === $arguments) {
```

Comment:
It will be better don't call this method if `$arguments === null`
Need to move this check out of method